scripts/social-banner/generate_rabbit_maximizer.py

```
@@ -0,0 +1,66 @@
+            return url
+    print("No icon URL found in project front matter", file=sys.stderr)
+    sys.exit(1)
+
+
+def download_icon(url):
+    """Download and decode an icon from *url*, returning a PIL Image."""
+    if urlparse(url).scheme not in ("http", "https"):
+        print(f"Refusing non-http(s) icon URL: {url}", file=sys.stderr)
+        sys.exit(1)
+    try:
+        data = urlopen(url, timeout=10).read()
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)